### PR TITLE
Cursor drag enhancement

### PR DESCRIPTION
**Pull request by ardiantutomo**
Hi @katspaugh, 

I create a small fix for this issue https://github.com/katspaugh/wavesurfer.ts/issues/77, Please take a look.

### CODE REVIEW

File: src/renderer.ts [hunk 1]
```diff
@@ -77,6 +78,7 @@ class Renderer extends EventEmitter<RendererEvents> {
   private initEvents() {
     // Add a click listener
     this.wrapper.addEventListener('click', (e) => {
+      if (this.isDragging) return
```
Comment (katspaugh) [inline src/renderer.ts:81]: Why is this needed?

File: src/wavesurfer.ts [hunk 2]
```diff
@@ -197,6 +197,8 @@ class WaveSurfer extends Player<WaveSurferEvents> {
       this.renderer.on('drag', (relativeX) => {
         if (this.options.interact) {
           const newTime = this.getCurrentTime() + this.getDuration() * relativeX
+          const progress = newTime / this.getDuration()
+          this.renderer.renderProgress(progress, this.isPlaying())
```
Comment (katspaugh) [inline src/wavesurfer.ts:201]: Does this make it smoother for you? In my tests it didn't.

**Review by katspaugh — COMMENTED** [review]
Hey, @ardiantutomo, thanks for the PR!
I have a couple questions.

File: src/renderer.ts (hunk 1 above)
Comment (ardiantutomogdp) [reply src/renderer.ts:81]: without that flag, it will trigger 'click' event when dragging is finished. Because of that, the cursor will jump to the location that triggered by the 'click' event.

Comment (ardiantutomo) [reply src/renderer.ts:81]: without that flag, it will trigger 'click' event when dragging is finished. Because of that, the cursor will jump to the location that triggered by the 'click' event.

File: src/wavesurfer.ts (hunk 2 above)
Comment (ardiantutomo) [reply src/wavesurfer.ts:201]: yes, in your tests, is it looks the same as previous?

Comment (katspaugh) [reply src/wavesurfer.ts:201]: My tests were with a local file where it didn't make a difference. I've just read XyzAQWERTY's comment in #77 and tried with a remote file, which indeed makes a difference. 👍

File: src/renderer.ts (hunk 1 above)
Comment (katspaugh) [reply src/renderer.ts:81]: Isn't that OK though? The click will happen at the same place as the last drag position, so it should not matter?

Comment (ardiantutomo) [reply src/renderer.ts:81]: maybe you can try to remove the flag, and see what's the difference.
I don't know why the cursor jumping if the 'click' event was triggered

Comment (katspaugh) [reply src/renderer.ts:81]: Let's merge your PR then, and I'll see if I can figure out why the click is problematic on main.

**Review by katspaugh — APPROVED** [review]
Thank you, @ardiantutomo!
